assignments/prototype-refactor.js

```diff
@@ -7,3 +7,100 @@ Prototype Refactor
 2. Your goal is to refactor all of this code to use ES6 Classes. The console.log() statements should still return what is expected of them.
 
 */
+class GameObject{
+    constructor(options) {
+    this.createdAt = options.createdAt;
+    this.dimensions = options.dimensions;
+
+}
+    destroy() {
+        return `${this.name} was removed from the game.`;
+}
+}
+
+
+class CharacterStats extends GameObject{
+    constructor(options){
+    super(options);
+    this.healthPoints = options.healthPoints;
+    this.name = options.name;
+}
+    takeDamage() {
+        return `${this.name} took damage.`;
+}
+}
+
+
+
+class Humanoid extends CharacterStats{
+    constructor(options){
+        super(options);
+        this.faction = options.faction;
```

Comment:
notice it is called faction here, while in the console.log below it is called team.  I changed the console.log to reference faction and it works.